frontend/src/metabase-types/store/mocks/state.ts

```diff
@@ -19,25 +18,30 @@ import { createMockSettingsState } from "./settings";
 import { createMockSetupState } from "./setup";
 import { createMockUploadState } from "./upload";
 
-export const createMockState = (
-  opts?: Partial<State> | Partial<EnterpriseState> | Partial<SdkStoreState>,
-): State => ({
-  admin: createMockAdminState(),
-  app: createMockAppState(),
-  auth: createMockAuthState(),
-  currentUser: createMockUser(),
-  dashboard: createMockDashboardState(),
-  embed: createMockEmbedState(),
-  entities: createMockNormalizedEntitiesState(),
-  metabot: createMockMetabotState(),
-  parameters: createMockParametersState(),
-  qb: createMockQueryBuilderState(),
-  requests: createMockRequestsState(),
-  routing: createMockRoutingState(),
-  settings: createMockSettingsState(),
-  setup: createMockSetupState(),
-  upload: createMockUploadState(),
-  modal: null,
-  sdk: createMockSdkState(),
-  ...opts,
-});
+export function createMockState<S extends Pick<SdkStoreState, "sdk">>(
```

Comment:
Convert to a function so I could use function overload. This return the correct type depending on the passed opts. If `opts` contains `sdk` then the return type would be `StateWithSdk` otherwise it would be just `State`

Reply:
very nice! just realized you can do this to overload, usually I would do a conditional type (much worse)